### DISCUSSION TITLE
No longer working on ESP8266

### DISCUSSION THREAD

**Discussion by xuancong84** [miguelgrinberg/microdot#140] [Q&A]
The repo should make it clear whether it is designed for ESP8266 or ESP32 only.
For latest versions of micropython image (v1.20.0 and v1.19.1):
```
>>> import uasyncio
>>> from microdot_asyncio import Microdot
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
MemoryError: memory allocation failed, allocating 344 bytes
```
So it is no longer working.
Moreover, flashing bin/micropython onto ESP8266 does not work at all. So it is totally useless for ESP8266.

**Comment by miguelgrinberg** [comment]
How are you installing microdot on your ESP8266 board? On such a small board it is likely that the only way for it to fit is to freeze it in a custom firmware.

> Moreover, flashing bin/micropython onto ESP8266 does not work at all. So it is totally useless for ESP8266.

This is your own misunderstanding. The `bin/micropython` binary that is in this repository is the Unix version of micropython, and it is used to run the unit tests. If you need to flash micropython on your ESP8266 board, you should download official builds for this board from micropython.org.

**Reply by xuancong84** [reply]
Then you should make it clear in `README.md` that on ESP8266, due to small RAM (only about 32-64KB), it has to be put as frozen code. And the use of `bin/micropython`.

**Reply by rkompass** [reply]
I found that putting a `mpy-cross` - compiled version `microdot.mpy` into the file system of the esp8266 allows to run simple examples.
See my other discusssion resp. `gpio.py`.

**Comment by hagenbuch** [comment]
I seem to have the same or a similar error on a much bigger ESP32 so maybe there is a different reason for this bug?

**Reply by rkompass** [reply]
I don't understand what kind of bug you might have.

**Reply by xuancong84** [reply]
The reason is that a much bigger ESP32 does have more flash (e.g., 16MB), but it does not have bigger RAM (typically only about 320KB). The current Python code loader loads the entire `.py` file in one go. So if the Python file is too big to fit in RAM, then it will run out of memory. You can pre-compile your `.py` into `.mpy` which is much smaller, or you can freeze into your firmware (that uses 16MB flash) to prevent RAM memory overflow.
